rss-client/src/app/Quiz/components/cheater-warning/cheater-warning.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, OnInit, Input } from '@angular/core';
+import {MatDialog} from '@angular/material/dialog';
+import { CheaterService } from 'src/app/services/cheater.service';
+
+/**
+ * @title Warning dialog to display whenever the mouse leaves the browser window
+ */
+@Component({
+  selector: 'app-cheater-warning',
+  templateUrl: './cheater-warning.component.html',
+  styleUrls: ['./cheater-warning.component.css']
+})
+export class CheaterWarningComponent implements OnInit {
+  open: boolean
+  dialogRef
+  
+  constructor(public dialog: MatDialog, private cheaterService: CheaterService) {
+    this.cheaterService.leftTab.subscribe(e => {
+      if (e) {
+        if (!this.open) {
+          this.openDialog()
+        }
+      }
+    })
+    this.cheaterService.invalidated.subscribe(e => {
+      if (e) {
+        this.dialogRef.close()
+      }
+    })
```

Comment:
Not sure on this one but I know in August's demos and my light reading it is best practice to do this in the ngOnInit() function. 
It’s a common practice to use ngOnInit to perform initialization logic even if this logic doesn’t depend on DI, DOM or input bindings.